src/test/unit/mcmc/hmc/nuts/base_nuts_test.cpp

```diff
@@ -228,18 +237,19 @@ TEST(McmcNutsBaseNuts, rho_aggregation_test) {
   stan::callbacks::stream_logger logger(debug, info, warn, error, fatal);
 
   sampler.build_tree(3, z_propose,
-                     p_sharp_left, p_sharp_right, rho,
+                     p_sharp_begin, p_sharp_end, 
+                     rho, p_begin, p_end,
                      H0, 1, n_leapfrog, log_sum_weight,
                      sum_metro_prob, logger);
 
-  EXPECT_EQ(7, sampler.rho_values.size());
+  EXPECT_EQ(7 * 3, sampler.rho_values.size());
   EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(0));
   EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(1));
-  EXPECT_EQ(4 * init_momentum, sampler.rho_values.at(2));
+  EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(2));
   EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(3));
   EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(4));
-  EXPECT_EQ(4 * init_momentum, sampler.rho_values.at(5));
-  EXPECT_EQ(8 * init_momentum, sampler.rho_values.at(6));
+  EXPECT_EQ(2 * init_momentum, sampler.rho_values.at(5));
+  EXPECT_EQ(4 * init_momentum, sampler.rho_values.at(6));
```

Comment:
There should be 14 more of these checks since there are now 3x the number of rho_values

Reply:
There's only one aggregate rho exposed in the `build_tree` function.  The others needed for the new tests are constructed from this aggregate rho on the fly internally and hence not accessible for unit testing.  That internal construction could be repeated externally using the boundary momenta but then there's nothing to compare it to.

Reply:
I don't understand this. These values are being returned so why shouldn't they be tested?